### DISCUSSION TITLE
Fish does not color the autocomplete results for folder and file based on their type

### DISCUSSION THREAD

**Discussion by zhou13** [fish-shell/fish-shell#10781] [Ideas]
When I type `ls ~/.rye/<tab>` in zsh, I got 
<img width="496" alt="Screenshot 2024-10-12 at 11 28 31 PM" src="https://github.com/user-attachments/assets/6e5b89a2-fcf4-4b60-847b-88d769e62be2">

It is easy to distinguish between files and folders.

However, When I type `ls ~/.rye/<tab>` in zsh, I got 
<img width="619" alt="Screenshot 2024-10-12 at 11 29 21 PM" src="https://github.com/user-attachments/assets/8a3a047f-bae6-4709-9bf7-fb6428c09981">

It is much hard to distinguish the files from folder.

I wonder if there is a way to use the `zsh` auto-completion behavior in fish?

**Comment by krobelus** [comment]
Good idea thanks. I imagine when we do that we'll also want to remove the shared prefix from the rendering of each completion (which has been debated a few times), because both the shared prefix and filename arguments are underlined, so that might clash.
Since we nowadays always insert the shared prefix on the first tab, it might no longer provide as much value.

**Comment by zanchey** [comment]
Some discussion in https://github.com/fish-shell/fish-shell/issues/8618#issuecomment-1006602837

**Comment by theherk** [comment]
Bums me out that this isn't a thing. There is just no easy way to tell what is a file and what is a directory aside from the trailing slash. Nor can you tell what is executable (not even an asterisk).

<img width="764" height="200" alt="Screenshot 2025-12-17 at 15 20 15" src="https://github.com/user-attachments/assets/dba7dfdd-eec4-4d65-a7fd-dd1c85d4e5a1" />
